### PR TITLE
Add person student teacher class

### PR DESCRIPTION
**Pull request by Iptysam**
Create class Person with the following:
Instance vars: @id, @name, and @age.
Constructor with name, age, and parent_permission as parameter. name and parent_permission are optional and have default values of "Unknown" and true.
Getters for @id, @name, and @age.
Setters for @name and @age.
Private method of_age? that returns true if @age is greater or equal to 18 and false otherwise.
Public method can_use_services? that returns true if person is of age or if they have permission from parents.
Create class Student with the following:
Inherits from Person.
Constructor extends parent's constructor by adding @classroom and a parameter for it.
Method play_hooky that returns "¯\(ツ)/¯".
Create class Teacher with the following:
Inherits from Person.
Constructor extends parent's constructor by adding @specialization and a parameter for it.
Override can_use_services? so it always returns true.
Each class should be saved in a separate file.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,143 @@
...
+  <ul>
+    <li><a href="https://www.ruby-lang.org/en/">Ruby</a></li>
+  </ul>
+</details>
+
+### Key Features <a name="key-features"></a>
+
+- **[person.rb](person.rb)**
+- **[teacher.rb](teacher.rb)**
+- **[student.rb](student.rb)**
```
Comment (Olamarx) [inline README.md]: - [ ] Key feature means what are the key features of your app, not asking for link to the files, kindly add features, not links.

File: README.md [hunk 2]
```diff
@@ -1 +1,143 @@
...
+# 📖 Ruby Morse Code <a name="about-project"></a>
+
+**Ruby Morse Code** is a project which Add new students or teachers.
+Add new books.
+Save records of who borrowed a given book and when.
+## 🛠 Built With <a name="built-with"></a>
+
+### Tech Stack <a name="tech-stack"></a>
+
+> Describe the tech stack and include only the relevant sections that apply to your project.
```
Comment (Olamarx) [inline README.md]: - [ ] Kindly remove this dummy data

File: README.md [hunk 3]
```diff
@@ -1 +1,143 @@
...
+
+### Setup
+
+Clone this repository to your desired folder:
+
+git clone git@github.com:Iptysam/School-library-OOP.git
+
+### Install
+
+Install this project with:
```
Comment (Olamarx) [inline README.md]: - [ ] With what? You must have missed that.

**Review by Olamarx — CHANGES_REQUESTED** [review]
Hi @Iptysam, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that does not have the readme set up professionally, kindly fix the comments I made as regards it.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by rloterh — COMMENTED** [review]
Hi Ipty, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

- Kindly ensure to remove the Microverse template logo link in your ReadMe, which appears broken. Optionally, you can replace it with your own logo if you have one.

- [NOT A REASON FOR INVALIDATION] Also, I noticed you have about triple _Person_ class, and no other required classes. I urge you to refactor your code so as to ensure the appropriate classes are in place 😃 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by frank1738 — APPROVED** [review]
Hi @Iptysam , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
✅ Correct use of Github workflow
✅ No linter error
✅ Organized code

## Optional suggestions

N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**